shared/reducers/pinentry.js

```diff
@@ -77,7 +77,7 @@ export default function (state: RootPinentryState = initialState, action: Pinent
       }
       return state
     default:
-      if (state.started && sessionID != null) {
+      if (state.started && sessionID != null && isPinentryAction(action)) {
```

Comment:
It's a really subtle bug.

Note that we return a modified state even if this isn't a pinentry action.

Next we don't pass in a proper initial state, just an empty object (which shouldn't type check, but does annoyingly. Can be fixed by making the type:
```
export type RootPinentryState = {
  started: boolean,
  pinentryStates: {
    [key: number]: ?PinentryState
  }
}

```
)

Then the final part of the bug is when we check to show the pinentry windows we look for the presence of `!state.closed` but since the pinentryState is just `{}` it passes that check.

This hasn't come up before because we've never passes the sessionID in the action.payload.sessionID.